build.sbt

```diff
@@ -6,14 +6,53 @@ lazy val commonSettings = Seq(
   scalacOptions ++= Seq("-deprecation","-unchecked","-Xsource:2.11"),
   libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.1" % "test",
   libraryDependencies += "org.json4s" %% "json4s-native" % "3.5.3",
+  libraryDependencies += "org.json4s" %% "json4s-jackson" % "3.5.3",
   libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value,
   addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
   resolvers ++= Seq(
     Resolver.sonatypeRepo("snapshots"),
     Resolver.sonatypeRepo("releases"),
     Resolver.mavenLocal))
 
-lazy val rocketchip = RootProject(file("rocket-chip"))
+val rocketChipDir = file("rocket-chip")
+
+lazy val firesimAsLibrary = sys.env.get("FIRESIM_IS_TOP") == None
+lazy val firesimDir = if (firesimAsLibrary) {
+  file("firesim/sim/")
+} else {
+  file("../../sim/")
+}
+
+// Subproject definitions begin
+// NB: FIRRTL dependency is unmanaged (and dropped in sim/lib)
+lazy val chisel  = (project in rocketChipDir / "chisel3")
+
+// Contains annotations & firrtl passes you may wish to use in rocket-chip without
+// introducing a circular dependency between RC and MIDAS
+lazy val midasTargetUtils = ProjectRef(firesimDir, "targetutils")
+
+// Rocket-chip dependencies (subsumes making RC a RootProject)
+lazy val hardfloat  = (project in rocketChipDir / "hardfloat")
+  .settings(
+    commonSettings,
+    crossScalaVersions := Seq("2.11.12", "2.12.4"))
+  .dependsOn(chisel, midasTargetUtils)
+
+
+lazy val macros     = (project in rocketChipDir / "macros")
+  .settings(commonSettings,
+  )
+
+// HACK: I'm strugging to override settings in rocket-chip's build.sbt (i want
```

Comment:
This is a gloriously hacky hack. This is obviously not something that you want to do, but I assume the reason you want midas target utils dependency is so you can go into rocket's source and add some debug things directly. SBT subprojects do seem to broken and hard to override, so this seems as good of a workaround as any.

The only other way I could see this going is a make step (or *shudders* sbt task) that does some surgery on the build.sbt in rocket before it gets loaded, but that seems more brittle and hacky to me.

Reply:
This sort of build.sbt modification really does have meta-build written all over it -- but i have no idea how to wield that power from a parent project.